ctl/filebuffer.go

```diff
@@ -0,0 +1,116 @@
+package ctl
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"os"
+	"sync"
+)
+
+func NewFileBuffer(max int, temp string) *FileBuffer {
```

Comment:
Include some godocs describing what the inputs mean.
```
// NewFileBuffer returns a file buffer which will use an in-memory buffer, until `max` bytes have been written, at which point it will write the contents of memory to a file, and continue writing future data to the file. The file will be written to `temp` directory.
```